src/index.ts

```diff
@@ -0,0 +1,76 @@
+import {IPC} from 'node-ipc';
+import {LanguageService} from 'typescript/lib/tsserverlibrary';
+
+import {SimpleDiagnostic} from './Protocol';
+
+function init(): ts.server.PluginModule {
+  function create(info: ts.server.PluginCreateInfo): LanguageService {
+    const ipc = new IPC();
+    ipc.config.id = 'ts-expose-status-plugin-server';
+    // We MUST silence logging, or else node-ipc will print to stdout. VSCode talks with the TS
+    // language service over stdin/stdout, so any extraneous messages to stdout will break VSCode's
+    // normal code intelligence features.
+    ipc.config.silent = true;
+
+    function log(s: string): void {
+      info.project.projectService.logger.info(s);
+    }
+
+    function convertToSimpleDiagnostic(diagnostic: ts.Diagnostic): SimpleDiagnostic {
+      return {
+        filePath: diagnostic.file ? diagnostic.file.fileName : null,
+        start: diagnostic.start != null ? diagnostic.start : null,
+        end:
+          diagnostic.start != null && diagnostic.length != null ? diagnostic.start + diagnostic.length : null,
+        message:
+          typeof diagnostic.messageText === 'string'
+            ? diagnostic.messageText
+            : diagnostic.messageText.messageText,
```

Comment:
It looks like this can be chained indefinitely: is it possible to call through to flattenDiagnosticMessageText ? https://github.com/Microsoft/TypeScript/blob/master/src/compiler/program.ts#L381

Reply:
The last messageText is guaranteed to be a string, according to the types, so it won't break. But sure, I'll see if I can I can find a way to call `flattenDiagnosticMessageText`.
```typescript
    interface DiagnosticRelatedInformation {
        category: DiagnosticCategory;
        code: number;
        file: SourceFile | undefined;
        start: number | undefined;
        length: number | undefined;
        messageText: string | DiagnosticMessageChain;
    }
...
    interface DiagnosticMessageChain {
        messageText: string;
        category: DiagnosticCategory;
        code: number;
        next?: DiagnosticMessageChain;
    }
```